### ISSUE TITLE
S3 bucket policy condition block displays endless diff with `values` array of length 1

### ISSUE THREAD

**Issue by crawforde** [hashicorp/terraform-provider-aws#20456]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform: v1.0.0
Provider: v3.52.0

### Affected Resource(s)

* aws_s3_bucket_policy

### Terraform Configuration Files

```hcl
resource "aws_s3_bucket" "foo" {
  bucket = "foo"
}

data "aws_iam_policy_document" "bar" {
  version = "2012-10-17"
  statement {
    sid = "ObjectWriteAccess"
    effect = "Allow"
    actions = [
      "s3:PutObject*"
    ]
    resources = [
      "${aws_s3_bucket.foo.arn}/*"
    ]
    principals {
      type        = "AWS"
      identifiers = [
        "arn:aws:iam::123456789012:user/alice@gmail.com",
        "arn:aws:iam::123456789012:user/bob@gmail.com"
      ]
    }
    condition {
      test = "StringEquals"
      variable = "s3:x-amz-acl"
      # Provider requires this value to be an array
      values = ["bucket-owner-full-control"]
    }
  }
}

resource "aws_s3_bucket_policy" "foo_bar" {
  bucket = aws_s3_bucket.foo.id
  policy = data.aws_iam_policy_document.bar.json
}

```
### Expected Behavior

After applying the configuration once, the expected behavior is that the bucket policy would show no diffs.

### Actual Behavior

The bucket policy shows a diff related to the condition block which does not go away after subsequent applications of the plan. The AWS API seems to be storing the value for `s3:x-amz-acl` in the condition block as a string rather than an array since there is only one value. The provider does not seem to be properly converting that value back into an array with one element on resource reads.

```
...
{
  ~ Condition = {
    ~ StringNotEquals = {
      ~ s3:x-amz-acl = "bucket-owner-full-control" -> [
        + "bucket-owner-full-control",
      ]
    } 
  }
  # (5 unchanged elements hidden)
}
...
Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

1. Use configuration in the example to create an s3 bucket (change the name) and an attached bucket policy. Create using `terraform apply`.
2. Run a subsequent `terraform apply`.

* #0000

**Comment by ewbankkit** [comment]
Potentially related: jen20/awspolicyequivalence#10.

**Comment by crawforde** [comment]
The last update to the `awspolicyequivalence` library was two years ago. Is there a plan to work around this if the library is stale?

**Comment by YakDriver** [comment]
@crawforde There is an effort: #22048

**Comment by mentos1386** [comment]
@YakDriver I'm using `v3.70.0` and still encountering this issue. Was it supposed to be fixed with #22048?

```tf
   ~ {
      ~ Condition = {
          ~ StringEquals = {
              ~ aws:PrincipalTag/bar = "foo" -> [
                  + "foo",
                ]
                # (1 unchanged element hidden)
            }
        }
        # (5 unchanged elements hidden)
    },
```
